rclcpp_lifecycle/include/rclcpp_lifecycle/lifecycle_service.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_LIFECYCLE__LIFECYCLE_SERVICE_HPP_
+#define RCLCPP_LIFECYCLE__LIFECYCLE_SERVICE_HPP_
+
+#include <memory>
+#include <string>
+
+#include "rclcpp/any_service_callback.hpp"
+#include "rclcpp/service.hpp"
+
+namespace rclcpp_lifecycle
+{
+/// base class with only
+/**
+ * pure virtual functions. A managed
+ * node can then deactivate or activate
+ * the service handling.
+ * It is more a convenient interface class
+ * than a necessary base class.
+ */
+class LifecycleServiceInterface
```

Comment:
is there any reason why we need a new interface and this cannot be the same as `LifecyclePublisherInterface`?
I would rather have a single `ManagedEntityInterface` than multiple ones.

Reply:
I thought of that too, but since it was a part that had to change the API that was not related to the purpose of this PR, I thought to make a PR that consolidates the interface and deprecates the existing API after this PR is included.

By the way, if #1846 is accepted, there may be no compatibility issues as that API will become an internal API.